domain/src/main/java/org/oppia/android/domain/util/StateRetriever.kt

```diff
@@ -191,17 +191,21 @@ class StateRetriever @Inject constructor() {
   private fun createVoiceoverMappingsFromJson(
     recordedVoiceovers: JSONObject
   ): Map<String, VoiceoverMapping> {
+    val voiceoverMappingMap = HashMap<String, VoiceoverMapping>()
     val voiceoverMappingJson = recordedVoiceovers.getJSONObject("voiceovers_mapping")
-    return voiceoverMappingJson.keys().asSequence().associateWith { contentId ->
+    voiceoverMappingJson.keys().asSequence().associateWith { contentId ->
```

Comment:
Could we maybe just add a filter on the map to filter out content IDs that map to empty voiceover mappings? Prefer not using a mutable map with a non-mutable map construction; this is a bit hard to follow and I suspect may be confusing in the future.

Reply:
PTAL at updated logic.